modules/programs/gnome-terminal.nix

```diff
@@ -6,6 +6,11 @@ let
 
   cfg = config.programs.gnome-terminal;
 
+  vteInitStr = ''
+    # gnome-terminal: Show current directory in the terminal window title.
+    type __vte_prompt_command &>/dev/null || . ${pkgs.gnome3.vte}/etc/profile.d/vte.sh
```

Comment:
Is the `type __vte_prompt_command` really needed? The init file should only be sourced once per shell and since it is not exported it should affect any sub-processes.

Reply:
I assume the same problem holds for the termite case.

Reply:
It might get sourced twice if one has both gnome-terminal and termite enabled. Not sure if that’s a big problem, though. Only thing that may happen is that zsh gets two `__vte_osc7` in its [`precmd_functions`](https://github.com/GNOME/vte/blob/master/src/vte.sh#L61).

Reply:
Ah, ok. Now I understand.

Yeah, ideally I suppose this should only be loaded if bash is actually running inside a VTE terminal. It might be worth creating an internal "programs.vte.enable" option that VTE based terminal modules could enable. It could then insert the `vte.sh` only once and condition its sourcing based on the presence of the `VTE_VERSION` environment variable. I know termite uses vte-ng but the vte.sh seems to be the same in both.